flexmeasures/data/models/planning/storage.py

```diff
@@ -645,14 +645,17 @@ def build_device_soc_values(
             )  # otherwise DST would be problematic
             if soc_datetime > end_of_schedule:
                 # Skip too-far-into-the-future target
+                disregarded_datetimes += [soc_datetime]
                 max_server_horizon = get_max_planning_horizon(resolution)
-                current_app.logger.warning(
-                    f"Disregarding target datetime {soc_datetime}, because it exceeds {end_of_schedule}. Maximum scheduling horizon is {max_server_horizon}."
-                )
                 continue
 
             device_values.loc[soc_datetime] = soc
 
+        if disregarded_datetimes:
+            current_app.logger.warning(
```

Comment:
Let's split off the case where the list has length 1, with a custom warning.

Reply:
made a specific message for a single case